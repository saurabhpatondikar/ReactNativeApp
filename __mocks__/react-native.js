const rn = require('react-native');
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter.js', () => {
  const {EventEmitter} = require('events');
  return EventEmitter;
});
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
module.exports = rn;
