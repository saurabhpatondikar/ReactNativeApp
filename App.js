/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 *
 */
// @flow
import React, {useEffect} from 'react';
import {addition} from './src/Reducers/StoreReducer';
import HomeScreen from './src/Screens/HomeScreen';
import AboutScreen from './src/Screens/AboutScreen';
import LastScreen from './src/Screens/LastScreen';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {applyMiddleware, createStore} from 'redux';
import {Provider} from 'react-redux';
import {ScreensName} from './src/Constants/ScreenName';
import thunk from 'redux-thunk';
import {NativeModules, Platform} from 'react-native';
const store = createStore(addition, applyMiddleware(thunk));
const Stack = createStackNavigator();
const AnModule = NativeModules.ReactJava;

const App = () => {
  useEffect(() => {
    if (Platform.OS === 'android') {
      AnModule.checkForEmulator();
    }
  }, []);
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen
            name={ScreensName.HomeScreen}
            component={HomeScreen}
            options={{headerShown: false}}
          />
          <Stack.Screen
            name={ScreensName.AboutScreen}
            component={AboutScreen}
            options={{headerShown: false}}
          />
          <Stack.Screen
            name={ScreensName.LastScreen}
            component={LastScreen}
            options={{headerShown: false}}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

export default App;
