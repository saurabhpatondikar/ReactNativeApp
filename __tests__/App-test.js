/**
 * @format
 */

import * as React from 'react';
import {render} from '@testing-library/react-native';
import App from '../App';

test('render App correctly or not', () => {
  const component = render(<App />);
  expect(component).toBeTruthy();
  expect(component).toMatchSnapshot();
});
