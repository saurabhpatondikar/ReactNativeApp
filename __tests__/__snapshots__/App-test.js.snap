// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App correctly or not 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#000000",
                            "flex": 1,
                            "padding": 10,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 0.5,
                            },
                            Object {
                              "justifyContent": "flex-start",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignSelf": "flex-end",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#2196F3",
                                  "fontSize": 16,
                                },
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                        <TextInput
                          allowFontScaling={true}
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          placeholder="Please Enter Your Name"
                          rejectResponderTermination={true}
                          style={
                            Array [
                              Object {
                                "alignSelf": "center",
                                "backgroundColor": "#ffffff",
                                "borderRadius": 10,
                                "borderWidth": 1,
                                "height": 56,
                                "marginTop": 15,
                                "width": "70%",
                              },
                            ]
                          }
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 0.5,
                            },
                            Object {
                              "padding": 5,
                            },
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "borderRadius": 10,
                              "height": 60,
                              "justifyContent": "center",
                              "margin": 5,
                              "opacity": 1,
                              "width": "80%",
                            }
                          }
                          testID="button"
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#2196F3",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#2196F3",
                                },
                              ]
                            }
                            testID="button_text"
                          >
                            Press Me
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#808080",
                              "borderRadius": 10,
                              "height": 60,
                              "justifyContent": "center",
                              "margin": 5,
                              "opacity": 1,
                              "width": "80%",
                            }
                          }
                          testID="button"
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#2196F3",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#2196F3",
                                },
                              ]
                            }
                            testID="button_text"
                          >
                            Press Me
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#2196F3",
                              "borderRadius": 10,
                              "height": 60,
                              "justifyContent": "center",
                              "margin": 5,
                              "opacity": 1,
                              "width": "80%",
                            }
                          }
                          testID="button"
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#2196F3",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#ffffff",
                                },
                              ]
                            }
                            testID="button_text"
                          >
                            Press Me
                          </Text>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "borderRadius": 50,
                                "borderWidth": 1,
                                "justifyContent": "center",
                                "margin": 5,
                              },
                              Object {
                                "backgroundColor": "#7CF3F9",
                                "borderColor": "#073436",
                                "width": "80%",
                              },
                            ]
                          }
                        >
                          <Text
                            importantForAccessibility=""
                            style={
                              Array [
                                Object {
                                  "alignSelf": "center",
                                  "position": "absolute",
                                },
                                Object {
                                  "color": "#083133",
                                  "fontSize": 16,
                                },
                              ]
                            }
                          >
                            Slide Me To Continue
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "height": 64,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
